mllib/src/main/scala/org/apache/spark/ml/tree/treeParams.scala

```diff
@@ -360,7 +362,8 @@ private[ml] trait RandomForestParams extends TreeEnsembleParams {
     "The number of features to consider for splits at each tree node." +
       s" Supported options: ${RandomForestParams.supportedFeatureSubsetStrategies.mkString(", ")}",
     (value: String) =>
-      RandomForestParams.supportedFeatureSubsetStrategies.contains(value.toLowerCase))
+      RandomForestParams.supportedFeatureSubsetStrategies.contains(value.toLowerCase)
+      || value.matches("^(?:[1-9]\\d*|0\\.\\d*[1-9]\\d*\\d*|1\\.0+)$"))
```

Comment:
This regex is repeated twice, does it perhaps make sense to move it to a constant (would have to be `private[spark]` probably to enable `mllib` package to read it). @sethah what do you think about that?